lib/Parse/Lexer.cpp

```diff
@@ -33,12 +34,6 @@
 
 #include <limits>
 
-// Regex lexing delivered via libSwift.
-static RegexLiteralLexingFn regexLiteralLexingFn = nullptr;
-void Parser_registerRegexLiteralLexingFn(RegexLiteralLexingFn fn) {
-  regexLiteralLexingFn = fn;
-}
```

Comment:
`lib/Parse` already depends on `ASTGen` (e.g. `swift_ASTGen_buildTopLevelASTNodes`) Instead of injecting  the function, just use the `@_cdecl` function in `ASTGen`